### DISCUSSION TITLE
Collison collisions with walls

### DISCUSSION THREAD

**Discussion by smalguy** [AlmasB/FXGL#1146] [Q&A]
Hello, 
me and some buddys are trying to make a RPG type game, similar to Zephyria.
We are curently starting to make the map, but on the Wiki there is only a tutorial on how to add collisons with Entitys and this left me wondering if there is any other way to create collisions, with walls and things that are not "entitys".

We would greatly appriciate some help or informatios on this topic, 
 samlguy.

**Comment by AlmasB** [comment]
Hi, you are possibly looking for something like this: [TopDownMoveComponent](https://github.com/AlmasB/FXGL/blob/3a1e7adb375e0369a6d156fac7e7c874de3b9053/fxgl/src/main/kotlin/com/almasb/fxgl/dsl/components/TopDownMoveComponent.kt). You don't have to use this specific component, but it should give you some ideas if you wish to have non-cell based movement.

For cell-based movement, use `CellMoveComponent` or if you also need pathfinding, then `AStarMoveComponent`.

Overall, depends on the use case. Hope this helps.
